### PR TITLE
Add Smooth Animations Plugin

### PR DESCRIPTION
**Pull request by DevDennis**
- Adds Smooth(interpolated) Animations for Players, NPCs, Objects and SpotAnims

image

Before:

After:

Before:

After:

### CODE REVIEW

**Comment by alex-vincent** [comment]
I've been testing this out, and it looks fantastic. Nice job!

**Comment by ShaunDreclin** [comment]
Sexy!

File: runelite-mixins/src/main/java/net/runelite/mixins/RSDynamicObjectMixin.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+		}
+	}
+
+	@FieldHook("animCycleCount")
+	@Inject
+	public void onAnimCycleCountChanged(int idx)
+	{
+		boolean fromConstructor = true;
+		// hack to see if this got called from the constructor
+		StackTraceElement[] stackTrace = new Throwable().getStackTrace();
```
Comment (devLotto) [inline runelite-mixins/src/main/java/net/runelite/mixins/RSDynamicObjectMixin.java]: Did you check how often this is run and if constructing this stracktrace will have an effect on performance/fps?

Comment (DevDennis) [reply runelite-mixins/src/main/java/net/runelite/mixins/RSDynamicObjectMixin.java]: I've added the license files and replaced the constructor check

**Comment by DevDennis** [comment]
Added some comments and verified this still works for the current revision

**Pull request merged**